### PR TITLE
FEAT: support pivot_table op

### PR DESCRIPTION
**Pull request by Hank0626**
## What do these changes do?
- support pivot_table op of dataframe
- Add unit test

## Check code requirements
- [x] tests added / passed (if needed)
- [x] Ensure all linting tests pass

### CODE REVIEW

File: python/xorbits/_mars/dataframe/base/pivot_table.py [hunk 1]
```diff
@@ -0,0 +1,433 @@
...
+    Parameters
+    ----------
+    values : list-like or scalar, optional
+        Column or columns to aggregate.
+    index : column, Grouper, array, or list of the previous
+        If an array is passed, it must be the same length as the data. The
+        list can contain any of the other types (except list).
+        Keys to group by on the pivot table index.  If an array is passed,
+        it is being used as the same manner as column values.
+    columns : column, Grouper, array, or list of the previous
```
Comment (UranusSeven) [inline python/xorbits/_mars/dataframe/base/pivot_table.py:274]: Ditto.

File: python/xorbits/_mars/dataframe/base/pivot_table.py [hunk 2]
```diff
@@ -0,0 +1,433 @@
...
+    Create a spreadsheet-style pivot table as a DataFrame.
+
+    The levels in the pivot table will be stored in MultiIndex objects
+    (hierarchical indexes) on the index and columns of the result DataFrame.
+
+    Parameters
+    ----------
+    values : list-like or scalar, optional
+        Column or columns to aggregate.
+    index : column, Grouper, array, or list of the previous
```
Comment (UranusSeven) [inline python/xorbits/_mars/dataframe/base/pivot_table.py:269]: Raise NotImplementedError on array and Grouper or list of them.

File: python/xorbits/_mars/dataframe/base/__init__.py [hunk 3]
```diff
@@ -78,6 +79,7 @@ def _install():
         setattr(t, "drop", df_drop)
         setattr(t, "pop", df_pop)
         setattr(t, "pivot", df_pivot)
+        setattr(t, "pivot_table", df_pivot_table)
```
Comment (UranusSeven) [inline python/xorbits/_mars/dataframe/base/__init__.py:82]: `pivot_table` is also a module member of `python/xorbits/_mars/dataframe/__init__.py`

File: python/xorbits/_mars/dataframe/base/pivot_table.py [hunk 4]
```diff
@@ -0,0 +1,433 @@
...
+    index = AnyField("index", default=None)
+    columns = AnyField("columns", default=None)
+    aggfunc = AnyField("aggfunc", default="mean")
+    fill_value = AnyField("fill_value", default=None)
+    margins = BoolField("margins", default=False)
+    dropna = BoolField("drapna", default=True)
+    margins_name = StringField("margins_name", default="All")
+    observed = BoolField("observed", default=False)
+    sort = BoolField("sort", default=True)
+    shuffle_size = Int32Field("shuffle_size")
```
Comment (aresnow1) [inline python/xorbits/_mars/dataframe/base/pivot_table.py:50]: Anywhere to set `shuffle_size`'s value?

File: python/xorbits/_mars/dataframe/base/pivot_table.py [hunk 5]
```diff
@@ -0,0 +1,433 @@
...
+    all_dtype_index = ListField("all_dtype_index", default=None)
```
Comment (aresnow1) [inline python/xorbits/_mars/dataframe/base/pivot_table.py:51]: `all_dtype_index` -> `output_columns`

File: python/xorbits/_mars/dataframe/base/pivot_table.py [hunk 6]
```diff
@@ -0,0 +1,433 @@
...
+        out = op.outputs[0]
+
+        for dtype in op.all_dtype_index:
+            if dtype not in input_data.dtypes.index:
+                input_data[dtype] = np.nan if op.fill_value is None else op.fill_value
+
+        ctx[out.key] = input_data
+
+    @classmethod
+    @redirect_custom_log
```
[the remark this answers is not in the document]
Comment (Hank0626) [reply python/xorbits/_mars/dataframe/base/pivot_table.py:92]: `pivot_table` can accept a user-defined function input `aggfunc`

File: python/xorbits/_mars/dataframe/base/pivot_table.py [hunk 7]
```diff
@@ -0,0 +1,433 @@
...
+
+        filtered_chunks = [chunk for chunk in out_chunks if chunk.shape[1] != 0]
+        for i, chunk in enumerate(filtered_chunks):
+            chunk._index = (i, 0)
+
+        all_dtype_index = set().union(
+            *[set(chunk.dtypes.index) for chunk in filtered_chunks]
+        )
+        dtypes_chunks = pd.concat([chunk.dtypes for chunk in filtered_chunks])
+        dtypes_chunks = dtypes_chunks.groupby(dtypes_chunks.index).first()
```
Comment (aresnow1) [inline python/xorbits/_mars/dataframe/base/pivot_table.py:212]:
```suggestion
        combined_dtypes = pd.concat([chunk.dtypes for chunk in filtered_chunks]).to_dict()
        output_columns = list(combined_dtypes.keys())
        output_dtypes = pd.Series(combined_dtypes)
```

File: python/xorbits/_mars/dataframe/base/pivot_table.py (hunk 4 above)
Comment (Hank0626) [reply python/xorbits/_mars/dataframe/base/pivot_table.py:50]: L159

File: python/xorbits/_mars/dataframe/base/pivot_table.py (hunk 5 above)
Comment (Hank0626) [reply python/xorbits/_mars/dataframe/base/pivot_table.py:51]: done

File: python/xorbits/_mars/dataframe/base/pivot_table.py (hunk 2 above)
Comment (Hank0626) [reply python/xorbits/_mars/dataframe/base/pivot_table.py:267]: done

File: python/xorbits/_mars/dataframe/base/pivot_table.py (hunk 1 above)
Comment (Hank0626) [reply python/xorbits/_mars/dataframe/base/pivot_table.py:272]: done

File: python/xorbits/_mars/dataframe/base/__init__.py (hunk 3 above)
Comment (Hank0626) [reply python/xorbits/_mars/dataframe/base/__init__.py:82]: done

**Review by aresnow1 — APPROVED** [review]

**Pull request merged**